app.js

```diff
@@ -8,6 +8,7 @@ const express = require('express'),
   errors = require('./app/middlewares/errors'),
   migrationsManager = require('./migrations'),
   logger = require('./app/logger'),
+  session = require('client-sessions'),
```

Comment:
You implemented a nice way of session and cookies authentication, but the card asked for a token based authentication.
Here is a relative short blog post where its explained the different type of authenticacion methods. https://hackernoon.com/how-do-you-authenticate-mate-f2b70904cc3a
Also if you wanna read a little more you can read this one, where they do a more extense comparison between cookies and tokens.
https://auth0.com/blog/cookies-vs-tokens-definitive-guide/